src/Camelot.Services.Windows/ShellIcons/WindowsIconTypes.cs

```diff
@@ -0,0 +1,26 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+
+namespace Camelot.Services.Windows.ShellIcons
+{
+    public class WindowsIconTypes
+    {
+        public static bool IsIconThatRequiresExtract(string filename)
+        {
+            if (string.IsNullOrEmpty(filename))
+                throw new ArgumentNullException(nameof(filename));
+
+            if (filename.Contains("?"))
+                return true; // icon in indexed resource
+            if (filename.Contains(","))
+                return true; // icon in indexed resource
+
+            var ext = Path.GetExtension(filename).ToLower();
+            var extensionsOfIcoFiles = new List<string>() { ".ico", ".exe", ".dll" , ".cpl", ".appref-ms", ".msc" };
```

Comment:
I suspect that this list should be in sync with `WindowsShellIconsService` and this code might be refactored to avoid this class creation at all